src/Common/src/System/Collections/CompatibleComparer.cs

```diff
@@ -17,10 +17,6 @@ internal CompatibleComparer(IHashCodeProvider hashCodeProvider, IComparer compar
             _comparer = comparer;
         }
 
-        internal IHashCodeProvider HashCodeProvider => _hcp;
```

Comment:
I think splitting the file into two just because of these two trivial properties is not an improvement. I would keep it as it is.

Reply:
This was suggested to me, but I agree with you. Will fix.